1350 hp data card

michael-barbour opened the pull request:
Closes #1350 

Example doc at http://127.0.0.1:8000/datadocument/53/

This ticket lands in a slightly awkward place.  None of the button are working yet as those are in #1345.  This really only shows that the card is created.

New requirement "factory-boy" added and tested by building production pods.

Fixtures changed to add tags.  In order to see them run `nuke_and_pave`

zachmunzenrider commented:
Getting one test failure:

```Failure
Traceback (most recent call last):
  File "C:\dev\factotum\dashboard\tests\functional\test_habits_and_practices.py", line 41, in test_hp_tags
    self.assertEqual(hp1.tags.first().updated_at.day, date.today().day)
AssertionError: 8 != 13```

michael-barbour commented:
> This is looking good, but I think the tag/tag kind isn't set up like Kathie wants. In the ticket, all HP tags associated with the data record (displayed as Kind: tag title); HP tags should be grouped by the HP tag 'kind'
> 
> Right now you've got the tags displaying as tag title: description. I'm guessing that the kinds should be things like age, SES, and race, and the tags themselves would be things like "adult" or "infant", with "age" as the kind
> 
> Also I'm not seeing where the last bullet is implemented:
> PUCs linked to that extractedhabitsandpractices record (displayed where??)
> 
> The factory boy code looks great! I look forward to using that in an upcoming ticket. . . .

Looks like the above request was added to the issue and not the PR.  I reworked the tag to now use Kind - Name instead of Name - Definition and I've also added a section to work as a PUC link section as well as working it into the tests.

I also merged in John Deere because I had to add a migration.
